### PR TITLE
added setdiff1d to jax.numpy

### LINKED ISSUE #22153: setdiff1d

### PR DESCRIPTION
**Pull request by ShreyanshBardia**
closes #22153

### CODE REVIEW

**Comment by ShreyanshBardia** [comment]
Hey @Mr-Niraj-Kulkarni, can you please take a look at this PR

**Comment by ShreyanshBardia** [comment]
Hi @AnnaTz, actually the main aim of the PR was to add `setdiff1d` which was depended on j`ax.numpy.where`, `jax.numpy.unique`  and `jax.numpy.in1d`. Since `in1d` wasn't implemented, I added that and then while testing 'setdiff1d' I found bugs in `where` and `unique` function which I tried fixing. Apologies for the mess, I'll create separate PRs for each of them and will keep this on standby until they are merged. Thanks!

**Comment by AnnaTz** [comment]
> Hi @AnnaTz, actually the main aim of the PR was to add `setdiff1d` which depends on `jax.numpy.where`, `jax.numpy.unique` and `jax.numpy.in1d`. Since `in1d` wasn't implemented, I added that and then while testing `setdiff1d` I found bugs in `where` and `unique` function which I tried fixing. Apologies for the mess, I'll create separate PRs for each of them and will keep this on standby until they are merged. Thanks!

I don't see how it depends on these frontend functions, since it is only calling ivy functions in its implementation.

**Comment by ShreyanshBardia** [comment]
The current implementation does depend on the frontend functions, the respective lines are mentioned below. 
https://github.com/unifyai/ivy/blob/d970ce9e5e220bd766b66bb4664ba1f3f99592e4/ivy/functional/frontends/jax/numpy/creation.py#L293
https://github.com/unifyai/ivy/blob/d970ce9e5e220bd766b66bb4664ba1f3f99592e4/ivy/functional/frontends/jax/numpy/creation.py#L304
There was also a bug in `ivy.full` which I tried fixing (now have made a separate PR), which was being used below
https://github.com/unifyai/ivy/blob/d970ce9e5e220bd766b66bb4664ba1f3f99592e4/ivy/functional/frontends/jax/numpy/creation.py#L286

**Comment by ShreyanshBardia** [comment]
I have mentioned below the respective PRs and their issues for a better understanding

`where`
Issue - #22356       PR -      #22357
`ivy.full` 
Issue - #22356      PR -       #22404

**Comment by AnnaTz** [comment]
> The current implementation does depend on the frontend functions, the respective lines are mentioned below.
> 
> https://github.com/unifyai/ivy/blob/d970ce9e5e220bd766b66bb4664ba1f3f99592e4/ivy/functional/frontends/jax/numpy/creation.py#L293
> 
> 
> https://github.com/unifyai/ivy/blob/d970ce9e5e220bd766b66bb4664ba1f3f99592e4/ivy/functional/frontends/jax/numpy/creation.py#L304
> 
> 
> There was also a bug in `ivy.full` which I tried fixing (now have made a separate PR), which was being used below
> https://github.com/unifyai/ivy/blob/d970ce9e5e220bd766b66bb4664ba1f3f99592e4/ivy/functional/frontends/jax/numpy/creation.py#L286

Oh right, don't know how I missed that 😅

**Comment by AnnaTz** [comment]
ivy-gardener

File: ivy/functional/frontends/jax/numpy/creation.py [hunk 1]
```diff
@@ -277,8 +277,33 @@ def zeros(shape, dtype=None):
...
+            n_unique = ar1.size + 1 - (ar1 == ar1[0]).sum(dtype=ivy.int64)
+            mask = ivy.where(ivy.arange(ar1.size) < n_unique, mask, False)
+        return ivy.where(
+            ivy.arange(size) < mask.sum(dtype=ivy.int64),
+            ar1[jnp_frontend.where(mask, size=size)[0].ivy_array],
+            fill_value,
+        )
+
+      
 def zeros_like(a, dtype=None, shape=None):
```
Comment (AnnaTz) [inline ivy/functional/frontends/jax/numpy/creation.py:307]: Hi @ShreyanshBardia, I think you have messed `zeros_like`'s decorators. Could you check on this?
Other than that, looks good and we can merge.

**Review by AnnaTz — CHANGES_REQUESTED** [review]

**Comment by KareemMAX** [comment]
ivy-gardener

**Comment by AnnaTz** [comment]
ivy-gardener

File: ivy/functional/frontends/jax/numpy/creation.py (hunk 1 above)
Comment (ShreyanshBardia) [reply ivy/functional/frontends/jax/numpy/creation.py]: Oh yes, this is a mistake. Reverted now, thanks!

**Review by AnnaTz — APPROVED** [review]
Great, thanks!

**Pull request merged**